traffic_portal/app/src/common/modules/table/certExpirations/TableCertExpirationsController.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var TableCertExpirationsController = function(tableName, certExpirations, dsXmlToIdMap, filter, $document, $scope, $state, $filter, locationUtils) {
+
+	/** All of the expiration fields converted to actual Dates */
+	$scope.certExpirations = certExpirations.map(
+		function(x) {
+			// need to convert this to a date object for ag-grid filter to work properly
+			x.expiration = new Date(x.expiration);
+			return x;
+		});
+
+	$scope.dsXmlToIdMap = dsXmlToIdMap;
+
+	$scope.editCertExpirations = function(dsId) {
+		locationUtils.navigateToPath('/delivery-services/' + dsId + '/ssl-keys');
+	}
+
+	/**
+	 * Formats the contents of a 'federation' column cell as "True" or blank for visibility.
+	 */
+	function federatedCellFormatter(params) {
+		if (!params.value) {
+			return '';
+		} else {
+			return params.value;
+		}
+	}
+
+	/** The columns of the ag-grid table */
+	$scope.columns = [
+		{
+			headerName: "Delivery Service",
+			field: "deliveryservice",
+			hide: false
+		},
+		{
+			headerName: "CDN",
+			field: "cdn",
+			hide: false
+		},
+		{
+			headerName: "Provider",
+			field: "provider",
+			hide: false
+		},
+		{
+			headerName: "Expiration",
+			field: "expiration",
+			hide: false,
+			filter: "agDateColumnFilter"
+		},
+		{
+			headerName: "Federated",
+			field: "federated",
+			hide: false,
+			valueFormatter: federatedCellFormatter
+		},
+	];
+
+	/** Options, configuration, data and callbacks for the ag-grid table. */
+	$scope.gridOptions = {
+		onRowClick: function(params) {
+			const selection = window.getSelection().toString();
+			if(selection === "" || selection === $scope.mouseDownSelectionText) {
```

Comment:
`$scope.mouseDownSelectionText` is either undefined or the empty string, and `window.getSelection.toString()` will always return a string. when `$scope.mouseDownSelectionText` is an empty string, the second part of this condition is redundant. `undefined` and `string` types have no overlap, so this condition will always evaluate to `false` in the case that `$scope.mouseDownSelectionText` is `undefined`.. Furthermore, the empty string is the only false-y string value, so all in all this condition can be written as simply `if (!selection) {`.